### DISCUSSION TITLE
Is it ok to have two properties marked with @PresentationState in the same state?

### DISCUSSION THREAD

**Discussion by ghost** [pointfreeco/swift-composable-architecture#2235] [Q&A] [answered]
Hello,

In our app, we'd like to treat alerts (of the kind you can just acknowledge with an OK button and do nothing else) separately from bona-fide presentations (sheets and stack presentations). Is it ok, then, to have something like
```swift
struct State: Equatable {
   // other state
   @PresentationState var destination: Destination.State?
   @PresentationState var alertState: AlertState<Never>?
}
```
(with the corresponding actions and all the rest that go along with `@PresentationState`) or will the two properties interfere with each other in how TCA handles presentation internally?

Should we do this, instead?
```swift
struct State: Equatable {
   // other state
   @PresentationState var destination: Destination.State?
   var alertState: AlertState<Never>?
}
```

Many thanks in advance.

**Comment by mbrandonw** [comment, marked as answer]
You can use as many `@PresentationState`'s as you want. Just remember to have the corresponding `PresentationAction` and `ifLet`.

**Reply by ghost** [reply]
Awesome. Thanks, Brandon. Your quick response is much appreciated.
